develop/smart-contracts/dev-environments/hardhat.md

```diff
@@ -0,0 +1,309 @@
+
+When writing tests, you should avoid using helpers like `time` and `loadFixture`. For example, for the `Lock.sol` contract, you can replace the default test file under `tests/Lock.js` with the following logic:
+
+```javascript title="Lock.js"
+--8<-- "code/develop/smart-contracts/dev-environments/hardhat/lock-test.js"
+```
+
+To run your test, execute the following command:
+
+```bash
+npx hardhat test
+```
+
+## Deploying with a Local Node
+
+Before deploying to a live network, you can deploy your contract to a local node using the [`hardhat-revive-node`](https://www.npmjs.com/package/hardhat-revive-node){target=\_blank} plugin and Ignition modules:
+
+1. First, ensure that you have compiled a Substrate node and the ETH RPC adapter from the Polkadot SDK. Checkout the [compatible commit](https://github.com/paritytech/polkadot-sdk/commit/c29e72a8628835e34deb6aa7db9a78a2e4eabcee){target=\_blank} from the SDK and build the node and the ETH-RPC from source:
+
+    ```bash
+    git clone https://github.com/paritytech/polkadot-sdk.git
+    cd polkadot-sdk
+    git checkout {{ dependencies.repositories.polkadot_sdk_compatible_hardhat_node }}
+    ```
+
+    Now, build the node and the ETH-RPC adapter. Note that this process might take a long time to complete:
+
+    ```bash
+    # Build the substrate node
+    cargo build --release
+    # Build the eth-rpc adapter
+    cargo build -p pallet-revive-eth-rpc --bin eth-rpc --release
+    ```
+
+2. Update the Hardhat configuration file to add the local node as a target for local deployment:
+
+    === "Remix Configuration"
+
+        ```javascript title="hardhat.config.js" hl_lines="27-44"
+        --8<-- 'code/develop/smart-contracts/dev-environments/hardhat/hardhat.config.js:0:12'
+        --8<-- 'code/develop/smart-contracts/dev-environments/hardhat/hardhat.config.js:15:29'
+          --8<-- 'code/develop/smart-contracts/dev-environments/hardhat/hardhat.config.js:43:59'
+        --8<-- 'code/develop/smart-contracts/dev-environments/hardhat/hardhat.config.js:65:66'
+        ```
+
+    === "Binary Configuration"
+
+        ```javascript title="hardhat.config.js" hl_lines="27-44"
+        --8<-- 'code/develop/smart-contracts/dev-environments/hardhat/hardhat.config.js:0:12'
+        --8<-- 'code/develop/smart-contracts/dev-environments/hardhat/hardhat.config.js:15:17'
+          --8<-- 'code/develop/smart-contracts/dev-environments/hardhat/hardhat.config.js:31:42'
+          --8<-- 'code/develop/smart-contracts/dev-environments/hardhat/hardhat.config.js:43:59'
+        --8<-- 'code/develop/smart-contracts/dev-environments/hardhat/hardhat.config.js:65:66'
+        ```
+
+    Ensure to replace to replace the `INSERT_PATH_TO_SUBSTRATE_NODE` and `INSERT_PATH_TO_ETH_RPC_ADAPTER` with the proper paths.
```

Comment:
Be more specific here or give an example where to find those files. Solidity developers may not be aware of how rust build system works and where to find the generated binaries